crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1873,6 +2038,20 @@ where
     }
 }
 
+fn aggregate_errors(
+    errors: &Vec<(SuiError, Vec<AuthorityName>, StakeUnit)>,
+) -> BTreeMap<String, (Vec<AuthorityName>, StakeUnit)> {
+    let mut aggregated_errors = BTreeMap::<String, (Vec<AuthorityName>, StakeUnit)>::new();
+    for (error, names, weight) in errors {
+        let entry = aggregated_errors
+            .entry(error.as_ref().to_string())
+            .or_default();
+        entry.0.extend_from_slice(names);
+        entry.1 += weight;
+    }
```

Comment:
nit: could destructure the entry to give its components names.

```suggestion
        let (validators, stake) = aggregated_errors
            .entry(error.as_ref().to_string())
            .or_default();
        validators.extend_from_slice(names);
        *stake += weight;
    }
```